Source/WebCore/platform/network/BlobRegistryImpl.cpp

```diff
@@ -261,19 +265,31 @@ void BlobRegistryImpl::registerInternalBlobURLForSlice(const URL& url, const URL
     addBlobData(url.string(), WTFMove(newData));
 }
 
-void BlobRegistryImpl::unregisterBlobURL(const URL& url, const std::optional<WebCore::SecurityOriginData>&)
+void BlobRegistryImpl::unregisterBlobURL(const URL& url, const std::optional<WebCore::SecurityOriginData>& topOrigin)
 {
     ASSERT(isMainThread());
-    if (m_blobReferences.remove(url.string()))
-        m_blobs.remove(url.string());
+    ASSERT(BlobURL::isInternalURL(url) || topOrigin);
+    auto& urlKey = url.string();
+    if (m_isPartitioningEnabled && m_blobs.contains(urlKey) && topOrigin && topOrigin != m_allowedBlobURLTopOrigins.get(urlKey)) {
```

Comment:
It seems check for `m_blobs.contains(urlKey)` is unnecessary because the operation can fail when the non-null origin is not on allowed list.

Reply:
Yes, good catch, I think that was from a previous attempt and I forgot to delete it.